deployments/static/custom-driver.yaml

```diff
@@ -0,0 +1,116 @@
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: nvidia-container-toolkit
+---
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  labels:
+    app: nvidia-container-toolkit-daemonset
+  name: nvidia-container-toolkit-daemonset
+spec:
+  selector:
+    matchLabels:
+      app: nvidia-container-toolkit-daemonset
+  template:
+    metadata:
+      labels:
+        app: nvidia-container-toolkit-daemonset
+    spec:
+      tolerations:
+        - key: nvidia.com/gpu
+          operator: Exists
+          effect: NoSchedule
+      priorityClassName: system-node-critical
+      serviceAccountName: nvidia-container-toolkit
+      hostPID: true
+      containers:
+      - image: nvidia/container-toolkit:1.15.0-ubuntu20.04
+        command: ["/work/nvidia-toolkit"]
+        # args:
+        #   - /bin/entrypoint.sh
+        env:
+        # Explicitly set NVIDIA_VISIBLE_DEVICES to void to prevent device injection.
+        - name: NVIDIA_VISIBLE_DEVICES
+          value: "void"
+        # Set the main arguments.
+        - name: ROOT
+          value: "/usr/local/nvidia"
+        - name: RUNTIME
+          value: "containerd"
+        # Set the toolkit installation arguments.
+        - name: NVIDIA_DRIVER_ROOT
+          value: /host/nvidia/driver
+        - name: DRIVER_ROOT_CTR_PATH
+          value: /driver-root
```

Comment:
Have you tested set it to /driver-root works to generate the correct CDI spec for lib path ? from my experiment, it seems like doesn't work well. 

From my understanding,  from the container, it has two places holding the driver:
1. `/driver-root`, as it is mounted to driver-install-path
2. `/host/</host/nvidia/driver>`, as it is mounted to </host/nvidia/driver> path 

Below code is doing the transform: https://github.com/NVIDIA/nvidia-container-toolkit/blob/973a6633b30a95e27aff4ef07772387aac961395/tools/container/toolkit/toolkit.go#L773 

If using the 2), it can generate the desired CDI spec path mount like `hostPath: "</host/nvidia/driver>/lib64/XXX", containerPath: "/lib64/XXX"`

if using the 1), it generate like `hostPath: "</host/nvidia/driver>/lib64/XXX", containerPath: "</host/nvidia/driver>/lib64/XXX"`, which is probably not what as we expected

Reply:
In my emulation of a GKE-like driver installation using kind it seems to work as expected. In the container when generating the spec I see:
```
time="2024-05-08T11:07:42Z" level=info msg="Selecting /driver-root/usr/lib/x86_64-linux-gnu/libcuda.so.515.105.01 as /usr/lib/x86_64-linux-gnu/libcuda.so.515.105.01"
``` 
(for example)

And in the generated management spec I see:
```
 grep libcuda.so management.nvidia.com-gpu.yaml
  - containerPath: /usr/lib/x86_64-linux-gnu/libcuda.so.515.105.01
    hostPath: /host/nvidia/driver/usr/lib/x86_64-linux-gnu/libcuda.so.515.105.01
```

Showing that the container path `/driver-root` is correctly transformed to the host path: `/host/nvidia/driver`.

I think the confusion may be that in the case I'm testing here, the node has the driver installed at `/host/nvidia/driver` -- which has nothing to do with `/host` in the toolkit contianer. I will update the example to use something that is less ambiguous.

Reply:
makes sense. Tested this change with GPU Operator changes by @cdesiniotis. It works as expected now.